governance/xc_admin/packages/xc_admin_frontend/pages/index.tsx

```diff
@@ -13,26 +14,28 @@ import { StatusFilterProvider } from '../contexts/StatusFilterContext'
 import { classNames } from '../utils/classNames'
 import '../mappings/signers.json'
 
+const readPublisherKeyToNameMapping = (filename: string) => {
+  if (fs.existsSync(filename)) {
```

Comment:
I would use the promisified version of this instead of using the sync version, the sync version is blocking and using the promisified version and turning this into an async function will enable you to use `Promise.all` to do the io in parallel

Reply:
yeah I thought of that but since it's only done once during startup and these are fairly small files (<30K) I thought it would be clearer to just take the minimal I/O hit.

Reply:
In general with modern node, there's not much of a good reason to prefer sync over async.  It's semantically incorrect and doesn't produce any better ergonomics since you could very easily use the async version here and just add `await` on line 34 and 35 and achieve pretty much the exact ergonomics you have here without the semantic incorrectness (it would still be blocking but at least it would be blocking in an obvious way at the callsite).

I don't know any good reason to use the sync functions nowadays.

Reply:
Oh actually I just remembered that `exists` has been [deprecated in node](https://nodejs.org/api/fs.html#fsexistspath-callback), one of the reasons is that doing this causes race conditions (obviously not in this case, but in the general case, a file could be removed between the call to `exists` and the call to `readFile`).

The more "correct" way to do this is to use `readFile` in a try-catch and handle `ENOENT` as a "file does not exist" error